### ISSUE TITLE
Add command line option for importing pdf

### ISSUE THREAD

**Issue by CosmicHorrorDev** [xournalpp/xournalpp#2204]
**Is your feature request related to a problem? Please describe.**

It would be nice to have an option for importing a pdf and saving as a `.xopp` file. I'm grading pdf submissions so I'm dealing with sets of 50+ separate pdfs at a time. For my workflow it would be a lot faster if I could go ahead and save all of them as a xopp file instead of having to import and save 50+ times.

**Describe the solution you'd like**

An application option along the lines of `xournalpp --import-pdf input.pdf output.pdf.xopp` or automatically naming the output to be `{input-filename}.xopp`

**Describe alternatives you've considered**

I guess some option to import all pdfs in a folder and save as a `.xopp` file in the GUI would be fine too, but I feel like that's not worth adding another UI option.

**Additional context**

**Comment by rolandlo** [comment]
That sounds like a good suggestion to me.

**Comment by beijoca** [comment]
I was looking for the opposite: running Xournalpp headless in order to batch convert .xopp files to .pdf, ideally recursively. 

(This would allow me to open and present all my existing journals on iPadOS, without having to export each and every file individually)

**Comment by rolandlo** [comment]
@beijoca That's already possible. Use the `xournalpp -p PDF XOPP` command, where you replace PDF by the name of the pdf to be created and XOPP for the name of your existing xopp file. 

Then e.g. the following bash script converts all the xopp-files in your folder to pdf-files where the pdf-filename is created by appending .pdf to the filename:

```
for XOPP in *.xopp 
do xournalpp -p $XOPP.pdf $XOPP
done
```
If you want to get other filenames, modify the script accordingly.

**Comment by beijoca** [comment]
@rolandlo Thank you very much!

**Comment by roblem** [comment]
I am in the exact same situation.  Around 100 assignments to grade and batch creation of the xopp would be a godsend.

**Comment by atticus-sullivan** [comment]
As a temporary workaround, you could try to write something with `xdotool` (with Linux) to automate the `.xopp` creation, but be careful to include enough time to wait until the next step otherwise bad things can happen.

**Comment by LovecraftianHorror** [comment]
@roblem So my temporary work-around for this is a basic bash script that handles opening pdfs. (This is assuming *nix although I'm sure similar can be done on other OSs)

So opening the pdf with `xournalpp` automatically opens it to annotate, so this is the (hacky) little script to batch open my pdfs called `xournal_open`.

```bash
#!/usr/bin/env bash

for file in "${@}"
do
    xournalpp "$file"&
    disown
done
```

and from there you can open selected pdfs with `xournal_open file1.pdf file2.pdf`. I put it in my `$PATH` and use this with `ranger` so it works well for me, but depending on your workflow if you have the RAM you could do `xournal_open *\.pdf` which will open all of them simultaneously (I'd love to do this since I grade one question at a time across all solutions, but my laptop can't take it) or just open the files one by one using `xournalpp` if that works well for you too. Just figured I would give you some options I've found over time!

**Comment by roblem** [comment]
@LovecraftianHorror  thanks for this.

Since by bash capabilities are so poor, I have some python functions I use for something similar (I apologize in advance if this is too much detail, but this seems to be a place where people are sharing tips). 

For everything below, I use `xournal_cmd = '/usr/bin/flatpak run com.github.xournalpp.xournalpp'`.  You would need to adust this for your os's xournalpp executable.

This one creates an xopp from all pdf's in a directory (this requires manual save for each one) for pdf annotation and should be the first thing your run:
```python
def instantiate_xopp(workdir, xournal_cmd):
    """
    For each pdf file in workdir, it checks that the xopp has been 
    instantiated for pdf annotation, and if not creates one (then 
    requires user to manually save).
    """
    xournal_cmd = xournal_cmd.split(' ')
    if workdir[-1]!='/':
        workdir = workdir +'/'
    for file in os.listdir(workdir):
        if file.endswith(".pdf"):
            check_name = file+'.xopp'
            if not os.path.exists(workdir+check_name):
                p = subprocess.Popen([*xournal_cmd, workdir+file], stderr=subprocess.DEVNULL)
                # wait until file closed by user
                p.wait()
    print("All files instantiated.")            
```
Once you run the above, this opens each xopp created above in random order for grading, displays progress in python repl (e.g. 67/94), and when you close an xopp it then opens the next one:
```python
def grade_xopp(workdir, xournal_cmd):
    """
    Randomly cycles through xopp's in workdir (setup for pdf annotations and created using instantiate_xopp) 
    in workdir.  You grade, save,  and close the xopp then it will open the next one. 

    Parameters
    ----------
    workdir : string
        Directory location of xopp files to grade

    Returns
    -------
    None.
    """
    xournal_cmd = xournal_cmd.split(' ')
    if workdir[-1]!='/':
        workdir = workdir +'/'
    files = os.listdir(workdir)
    random.shuffle(files)
    count = 0
    xopp_files = [f for f in files if f.endswith(".xopp")]
    totfiles = len(xopp_files)    
    for file in files:
        if file.endswith(".xopp"):
            print("Working on file %2.0f of %2.0f."%(count+1, totfiles))
            #pdb.set_trace()
            p = subprocess.Popen([*xournal_cmd, workdir+file], stderr=subprocess.DEVNULL)
            # wait until file closed by user
            p.wait()
            count+=1      
```
Finally, once grading is completed, this one exports the xopp's to a new pdf (no user intervention required).  I was getting strange errors with `subprocess` so had to use `os.system` call instead, but it works.  It also downsamples the pdf since some of these are
very large that for me caused blackboard to hang for long periods while it uploaded.
```python
def export_graded(inputdir, xournal_cmd, outputdir = None):
    """
    Parameters
    ----------
    inputdir : string
        location of xopp graded files

    outputdir: string
        if none: inputdir + 'graded'
        if not none: full path for graded excercises

    Returns
    -------
    """
    if inputdir[-1]!='/':
        inputdir = inputdir +'/'
    
    # create graded folder if outputdir is null
    if outputdir is None:
        path = inputdir + 'graded/'
    else:
        if outputdir[-1]!='/':
            outputdir = outputdir + '/'
        path = outputdir
    
    # check if path exists, and if not, create
    if not os.path.exists(path):
        os.mkdir(path)
    
    # all files in xopp directory
    files = os.listdir(inputdir)
        
    # cycle through xopp files in inputdir
    for file in files:
        if file.endswith(".xopp"):
            cname = clean_names(file)
            #pdb.set_trace()
            os.system(xournal_cmd + ' ' + '"' + inputdir+ file + '"' + ' -p ' + '"' + path+cname + '"')
            # some of these files are huge, use convert to downsample
            print("\n")
            z = subprocess.Popen(['/usr/bin/convert','-density','80x80','-quality','40','-compress','jpeg', path+cname, path+cname])
            print("the commandline is {}".format(subprocess.list2cmdline(z.args)))
```
and this is a utility function for creating better names (most of the time) than what blackboard gives you so that you can sort the graded assignments by last name making it easier to upload the graded assignment back to blackboard. - used by the export function above:            
``` python
def clean_names(name)
    """
    Parameters
    ----------
    workdir : string
        A xopp filename for downloaded bb assignment (e.g. MidTerm 1_mmwalker_attempt_2020-09-10-16-59-13.pdf)

    Returns
    -------
    names like this: walker_mm_graded.pdf

   Note: this function will assign incorrect names for students having usernames like wxu as it expects the first 
             two characters in the name blackboard assign to be first and middle initial. 
    """
    temp = name.split('_')
    assignment = temp[0]
    name = temp[1]
    lastname = temp[1][2:]
    initials = temp[1][:2]
    clean_name = lastname + '_' + initials + '_' + assignment.replace(" ", "_") + "_graded.pdf"
    clean_name_ = clean_name.strip('#')
    return clean_name_
```

**Comment by vlstill** [comment]
Hi. What is the status of this? I have the same use case (batch-exporting `.xopp` from `.pdf` so it can be edited more easily).

**Comment by piegamesde** [comment]
As a work-around, one could manually create such a file from the PDF. All we need is a list of all pages with their size. ([Reference](http://www-math.mit.edu/~auroux/software/xournal/manual.html#file-format)).

```xml
<?xml version="1.0" standalone="no"?>
<xournal creator="piegames" fileversion="4">
<title>Xournal++ document - see https://github.com/xournalpp/xournalpp</title>

<page width="612.00000000" height="792.00000000">
<background type="pdf" domain="absolute" filename="/path/test.pdf" pageno="1ll"/>
<layer/>
</page>

<page width="612.00000000" height="792.00000000">
<background type="pdf" pageno="2ll"/>
<layer/>
</page>

</xournal>
```

`domain` and `filename` attributes must be specified for the first page. I don't know what the `ll` suffix in the page numbers is for.

**Comment by ebits21** [comment]
Just wanted to chime in that I would love this to work from the command line in Windows.

I use Xournal++ at work to write up hearing test results. The problem is I have to manually make an .xopp file for each patient's pdf.

Would love to automate this! It's a lot of open > save every day!

**Comment by rolandlo** [comment]
This can easily be implement. Suggested syntax:
```term
xournalpp --save-document output.xopp input.pdf
```
I will open a PR soon.

**Comment by rolandlo** [comment]
There is now a PR available in #5313. It needs some polish, but it basically works. You can try it out with the binary generated there.
If you want to apply it to all `.pdf`-files in the current directory, use
```bash
for file in *.pdf; do xournalpp --save-document $file.xopp $file; done
```
on Linux. In my test it took about 1 second for 50 documents.

**Comment by ebits21** [comment]
Excellent. Works great on Windows once Xournal is added to Path.

Will save me a lot of time! 😃
